### PR TITLE
TaskManager.task_needs_computation

### PR DESCRIPTION
**Pull request by Wiezzel**
Refactored nested function into a public method. Used it in MaskUpdateService instead of Task.needs_computation(). This will prevent updating masks for timed-out tasks.

### CODE REVIEW

File: golem/task/taskmanager.py [hunk 1]
```diff
@@ -327,6 +327,16 @@ def got_wants_to_compute(self,
                                      op=TaskOp.WORK_OFFER_RECEIVED,
                                      persist=False)
 
+    def task_needs_computation(self, task_id: str) -> bool:
+        if self.tasks_states[task_id].status not in self.activeStatus:
+            logger.info(f'task is not active: {task_id}')
```
Comment (tworec) [inline golem/task/taskmanager.py:332]: @mfranciszkiewicz asked to put node info into logs. It helps debugging

**Review by tworec — CHANGES_REQUESTED** [review]

File: golem/task/taskmanager.py (hunk 1 above)
Comment (Wiezzel) [reply golem/task/taskmanager.py:332]: I can add another logging command in `get_next_subtask` but not here.

Comment (Wiezzel) [reply golem/task/taskmanager.py:332]: Ok, there you go.

Comment (tworec) [reply golem/task/taskmanager.py:332]: ok, but now task_id is redundantly twice...

**Review by tworec — CHANGES_REQUESTED** [review]

File: golem/task/taskmanager.py (hunk 1 above)
Comment (Wiezzel) [reply golem/task/taskmanager.py:332]: Better now?

**Comment by tworec** [comment]
selfie-1

**Review by tworec — APPROVED** [review]

File: golem/task/taskmanager.py (hunk 1 above)
Comment (jiivan) [reply golem/task/taskmanager.py:332]: Please use logging function parameters for string format arguments.

https://gist.github.com/jiivan/eedb505574ccb2503af48fa2e3043036

Comment (Wiezzel) [reply golem/task/taskmanager.py:332]: @jiivan Actually, you didn't include the f-string method in your logtest. I did and the difference for string argument is negligible:
```
#'message'
               correct_logging: 1069.55ms
             incorrect_logging: 1213.96ms
    newstyle_incorrect_logging: 1250.48ms
         format_string_logging: 1089.07ms
```
And using f-strings is more readable than using `%s`.

Comment (tworec) [reply golem/task/taskmanager.py:332]: interesting comparison, I think you can post it on #developers channel on our rocket. I also like f'string readability but thought that it is much slower than _'correct logging'_

Comment (tworec) [reply golem/task/taskmanager.py:332]: @Wiezzel create PR for @jiivan code with f'string ;)

Comment (jiivan) [reply golem/task/taskmanager.py:332]: @Wiezzel That's interesting, I didn't know about that. Maybe that's why linter didn't complain about f string method. Readability is subjective. I'll leave it up to you then.

Comment (tworec) [reply golem/task/taskmanager.py:332]: we can do a poll inside our team for readablity of f'strings and other string formatting methods !

Comment (Wiezzel) [reply golem/task/taskmanager.py:332]: I will leave it as it is. But we surely need some explicit, team-wide coding standards.

**Pull request merged**